src/conduits/RwaInputConduit3.sol

```diff
@@ -0,0 +1,183 @@
+    /**
+     * @notice `wad` amount of Dai was pushed to `to`
+     * @param to The RwaUrn address
+     * @param wad The amount of Dai
+     */
+    event Push(address indexed to, uint256 wad);
+
+    /**
+     * @notice Define addresses and gives `msg.sender` admin access.
+     * @param _dai Dai token contract address.
+     * @param _to RwaUrn contract address.
+     */
+    constructor(address _dai, address _gem, address _psm, address _to) public {
+        dai = DSTokenAbstract(_dai);
+        gem = DSTokenAbstract(_gem);
+        psm = PsmAbstract(_psm);
+        to = _to;
+
+        require(GemJoinAbstract(psm.gemJoin()).gem() == _gem, "RwaInputConduit3/wrong-gem-for-psm");
+        require(address(psm.dai()) == _dai, "RwaInputConduit3/wrong-dai-for-psm");
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "RwaInputConduit3/not-authorized");
+        _;
+    }
+
+    /**
+     * @notice Grants `usr` admin access to this contract.
+     * @param usr The user address.
+     */
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` admin access from this contract.
+     * @param usr The user address.
+     */
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    /**
+     * @notice Grants `usr` push access to this contract.
+     * @param usr The user address.
+     */
+    function mate(address usr) external auth {
+        may[usr] = 1;
+        emit Mate(usr);
+    }
+
+    /**
+     * @notice Revokes `usr` push access from this contract.
+     * @param usr The user address.
+     */
+    function hate(address usr) external auth {
+        may[usr] = 0;
+        emit Hate(usr);
+    }
+
+    /**
+     * @notice Internal method which:
+     *  - Approve PSM gemJoin Adapter with WAD amount of GEM
+     *  - Swap GEM to DAI through PSM
+     *  - Push DAI to Recipient address (to)
+     * @param wad Amount to swap and push
+     */
+    function _swapAndPush(uint256 wad) internal {
+        gem.approve(address(psm.gemJoin()), wad);
+        psm.sellGem(address(this), wad);
+
+        uint256 balance = dai.balanceOf(address(this));
+        dai.transfer(to, balance);
+
+        emit Push(to, balance);
+    }
+
+    /**
+     * @notice Method to swap WAD amount of USDC contract balance to DAI through PSM and push it to Recipient address (to).
+     * @dev `msg.sender` must first receive push acess through mate().
+     * @param wad Amount to swap and push
+     */
+     function push(uint256 wad) public {
+        require(may[msg.sender] == 1, "RwaInputConduit3/not-mate");
+        require(wad > 0, "RwaInputConduit3/not-positive-amount");
+        uint256 balance = gem.balanceOf(address(this));
+        require(balance >= wad, "RwaInputConduit3/insufficient-gem-balance");
+
+        _swapAndPush(wad);
+    }
```

Comment:
I think we should probably not add this method.
We we want to control the amount of Dai being disbursed, we should probably do it in the `urn.draw()` step, not in the conduit.